### PR TITLE
Shard ID in all outport structures

### PR DESCRIPTION
**Pull request by miiu96**
## Reasoning behind the pull request
- Added the shard ID in all the outport structures
  
## Proposed changes
- Changed the version of the `mx-chain-core` module (https://github.com/multiversx/mx-chain-core-go/pull/244)
- Populated `shardID` field in the outport structures
- Changed the version of the `mx-chain-es-indexer-go` module ( https://github.com/multiversx/mx-chain-es-indexer-go/pull/236)

## Testing procedure
- Verify if the explorer is working correctly 
- Verify if the log `WARN [2023-07-20 10:21:15.370]  [2/1084/15621812/(START_ROUND)] dataDispatcher.doWork could not index item (will retry) error =....` will no longer appears

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by sstanculeanu — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -14,7 +14,7 @@ require (
 	github.com/gorilla/websocket v1.5.0
 	github.com/mitchellh/mapstructure v1.5.0
 	github.com/multiversx/mx-chain-communication-go v1.0.4
-	github.com/multiversx/mx-chain-core-go v1.2.11
+	github.com/multiversx/mx-chain-core-go v1.2.12-0.20230725101425-a34bd6d3e3b9
```
Comment (bogdan-rosianu) [inline go.mod:17]: proper tag before merging

Comment (miiu96) [reply go.mod]: added

**Review by bogdan-rosianu — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Comment by gabi-vuls** [comment]
@@                            Log scanner                                  @@
##                            indexer-with-metrics                         ##
================================================================================
- Known Warnings              16
- New Warnings                6
- Known Errors                0
- New Errors                  0
- Panics                      0
================================================================================
- block hash does not match                  9612
- wrong nonce in block                       3783
- miniblocks does not match                  0
- num miniblocks does not match              0
- miniblock hash does not match              0
- block bodies does not match                0
- receipts hash missmatch                    0
================================================================================

**Pull request merged**